UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/EssWebhookController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.Services;
+using UKHO.ExchangeSetService.Common.Logging;
+using UKHO.ExchangeSetService.Common.Models.Request;
+
+namespace UKHO.ExchangeSetService.API.Controllers
+{
+    [ApiController]
+    [Authorize]
+    public class EssWebhookController : BaseController<EssWebhookController>
+    {
+        private readonly IEssWebhookService essWebhookService;
+
+        public EssWebhookController(IHttpContextAccessor contextAccessor,
+                                    ILogger<EssWebhookController> logger,
+                                    IEssWebhookService essWebhookService)
+        : base(contextAccessor, logger)
+        {
+            this.essWebhookService = essWebhookService;
+        }
+
+        [HttpOptions]
+        [Route("/newfilespublished")]
```

Comment:
Also, having slept on it, I think it would be best to start the path with `/webhook` please, i.e.:
```suggestion
        [Route("/webhook/newfilespublished")]
```
This is the pattern that other services have used for their webhook endpoints and I think it makes sense.

Reply:
Thanks, done as expected